### ISSUE TITLE
Detecting unneeded `async` keywords on functions

### ISSUE THREAD

**Issue by Greesb** [astral-sh/ruff#9951]
Hello,

I refactored a bunch of code and thought about a rule that could be useful: detecting function defined with the `async` keywords not using any `await` inside it, example:
```python

async def unneeded_async() -> None:  # should raise the error
    print("lol")

async def legit_async() -> None:  # should not raise the error
    await some_stuff()
```

There's probably some use cases where the async is needed, but, imo, most of the time those async can be removed.

What do you think ?

**Comment by Greesb** [comment]
Doesn't look like it is, the issue you mentioned is for detecting missing `await` keywords when calling `async` functions. 

What i'm talking about is to detect functions that have the `async` keyword but do not need it because there's no `await` inside them. @zanieb

**Comment by mikeleppane** [comment]
Yeah, this is not a direct duplicate of #9833. What can I think of a possible use case might be having a function that returns Coroutine/Awaitable and that return value is passed in asyncio.create_task.

**Comment by zanieb** [comment]
Yeah I see, it's a little mixed up over there as we're discussing both things.

We can track this separately.

@plredmond I'll move you over to this issue.

**Comment by plredmond** [comment]
In progress implementation: https://github.com/astral-sh/ruff/compare/main...plredmond:ruff:issue9951-spurious_async
